### ISSUE TITLE
[BUG] [2.10] Unable to create the CIS scan output because of the configMap size is exceeding 1 MB 

### ISSUE THREAD

**Issue by vardhaman22** [rancher/rancher#47296]
SURE-8850
**Rancher Server Setup**
- Rancher version: v2.8.5
- Installation option (Docker install/Helm Chart):
   - If Helm Chart, Kubernetes Cluster and version (RKE1, RKE2, k3s, EKS, etc): Docker
- Proxy/Cert Details:

**Information about the Cluster**
- Kubernetes version: 1.28.x
- Cluster Type (Local/Downstream): Downstream
   - If downstream, what type of cluster? (Custom/Imported or specify provider for Hosted/Infrastructure Provider):  Custom

**User Information**
- What is the role of the user logged in? (Admin/Cluster Owner/Cluster Member/Project Owner/Project Member/Custom) Admin
  - If custom, define the set of permissions:

**Describe the bug**

When executing the 1.8 hardened profile on RKE2 based cluster (Standard CIS operator installed via rancher apps), the scan never completes - and sticks in "running" state.
We have looked in the runner logs, and confirmed using the requests (k8s api) that the issue is that the "cisscan-output-" output (i.e. the report) fails to be created, because the resulting configMap is > 1 MB (etcd size limit). Logs show:
```

{"resource":"configmaps","namespace":"cis-operator-system","name":"cisscan-output-for-scan-p87fz","apiVersion":"v1"},"responseStatus":{"metadata":{},"status":"Failure","message":"ConfigMap \"cisscan-output-for-scan-p87fz\" is invalid: []: Too long: must have at most 1048576 bytes","reason":"Invalid","details":{"name":"cisscan-output-for-scan-p87fz","kind":"ConfigMap","causes":[{"reason":"FieldValueTooLong","message":"Too long: must have at most 1048576 bytes","field":"[]"}]},"code":422},
```
**To Reproduce**

1. create a hardened RKE2 cluster with 35 worker nodes and 5 controlplane and 5 etcd nodes.
2. Run a scan on it with rke2-cis-1.8-hardened profile

**Result**
scan will stuck in running state and security-scan-runner pod will fail with error like the one below
```

{"resource":"configmaps","namespace":"cis-operator-system","name":"cisscan-output-for-scan-p87fz","apiVersion":"v1"},"responseStatus":{"metadata":{},"status":"Failure","message":"ConfigMap \"cisscan-output-for-scan-p87fz\" is invalid: []: Too long: must have at most 1048576 bytes","reason":"Invalid","details":{"name":"cisscan-output-for-scan-p87fz","kind":"ConfigMap","causes":[{"reason":"FieldValueTooLong","message":"Too long: must have at most 1048576 bytes","field":"[]"}]},"code":422},
```
**Expected Result**

scan should be completed successfully.
**Screenshots**

**Additional context**

- similar issues were reported earlier:
https://github.com/rancher/rancher/issues/37200 and https://github.com/rancher/rancher/issues/41182

**Comment by vardhaman22** [comment]
the JSON file using which the configmap is created has a field `avmap` for each check, in which results of the audit commands are stored for all the nodes.
For some audit commands the output is quite large so for larger clusters this can cause the ConfigMap Size to exceed 1MB limit. This field is not used anywhere in rancher UI but the data is available in ClusterScanReport Object for a scan. 
Other static data i.e. audit text, audit command, remediation is well within the limit i.e. around 130 KB. And apart from that there are node names which are added in the report.

To solve this:
1. the avmap data of all the checks will be combined in an array.
2. the data will then be gzip compressed and base64 encoded into a string.
3. a new field inside the ClusterScanReport's reportJSON data will be introduced at top level with this string as value.

**Comment by rishabhmsra** [comment]
Validated on rancher v2.10-head

- Created RKE1, RKE2 and K3s v1.31 clusters.
- Installed CIS released version 6.3.0, and ran v1.8 permissive profile scan on all above clusters.
- Downloaded cluster scan report yaml.
- Upgraded chart to 7.0.0-rc.2 version.
- Ran the v1.8 permissive profile scans and all passed successfully.
- Also downloaded cluster scan report yaml.
- Compared the scan report yaml size and it's significantly reduced in CIS 7.0.0-rc.2 version.
- Cluster scans were also same for both 6.3.0 and 7.0.0-rc.2 chart versions.

**Comment by vardhaman-surana** [comment]
we will need to mention this command in the docs 
```
export REPORT="<scan-report-name>"
kubectl get clusterscanreport $REPORT -o json |jq ".spec.reportJSON | fromjson" | jq -r ".actual_value_map_data" | base64 -d | gunzip | jq .
```
